### PR TITLE
Support for installing solidity and single file things in general

### PR DESCRIPTION
**Pull request by mattgodbolt**

### CODE REVIEW

**Comment by mattgodbolt** [comment]
See https://github.com/compiler-explorer/compiler-explorer/pull/3507

File: bin/lib/installation.py [hunk 1]
```diff
@@ -1078,6 +1134,8 @@ def _targets_from(node, enabled, context, name, base_config):
     'nightlytarballs': NightlyTarballInstallable,
     'nightly': NightlyInstallable,
     'script': ScriptInstallable,
+    'solidity': SolidityInstallable,
```
Comment (partouf) [inline bin/lib/installation.py:1137]: was wondering if this name is appropriate, but seems so, its specific to the .json file and the solidity_available_releases() function

**Comment by partouf** [comment]
I'm in favor of cutting some versions, but I don't know anything about this language so I wouldn't be able to choose myself

**Comment by shekhirin** [comment]
> I'm in favor of cutting some versions, but I don't know anything about this language so I wouldn't be able to choose myself

I think it's not a big deal since versions are displayed in descending order from the most recent. I thought of use case when people want to compare how assembly for certain snippets change over time and it may be useful to have even old versions for that.

**Comment by RubenRBS** [comment]
Thanks for adding solidity support, we really appreciate you taking the time to open up the necessary PRs :)
As @partouf mentioned, there were lots of patch versions added, so I kept only the latest one for each minor version as a start.

We can always add more if there are specific ones needed/when new ones get released (As we don't remove compilers once we install them)

**Pull request merged**